version: '3'
services:
  reverse:
    image: traefik:v2.0
    ports: 
      - "443:443"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./letsencrypt:/letsencrypt
    command:
      - "--providers.docker=true"
      - "--entrypoints.websecure.address=:443"
      - "--certificatesresolvers.mytlschallenge.acme.tlschallenge=true"
      - "--certificatesresolvers.mytlschallenge.acme.email=malek.attia@pfechercheriattia.fr"
      - "--certificatesresolvers.mytlschallenge.acme.storage=/letsencrypt/acme.json"

  cells:
    image: pydio/cells:latest
    ports:
      - "8000:80"
    environment:
      - CELLS_BIND=cells.domain.tld:443
      - CELLS_EXTERNAL=https://cells.domain.tld
      - CELLS_NO_SSL=1
    labels:
      - "traefik.http.routers.cells.rule=Host(`pfechercheriattia.fr`)"
      - "traefik.http.routers.cells.entrypoints=websecure"
      - "traefik.http.routers.cells.tls.certresolver=mytlschallenge"

  mysql:
     image: mysql:5.7
     restart: always
     environment:
         MYSQL_ROOT_PASSWORD: P@ssw0rd
         MYSQL_DATABASE: cells
         MYSQL_USER: pydio
         MYSQL_PASSWORD: P@ssw0rd
     command: [mysqld, --character-set-server=utf8mb4, --collation-server=utf8mb4_unicode_ci]
     ports: ["3306:3306"]
    #extra_hosts:
    #  - "<cells.domain.tld>:<ip_to_access_domain>"
